### ISSUE TITLE
(t) data-collector (Dashboard backend) fails to start

### ISSUE THREAD

**Issue by phillxnet** [rockstor/rockstor-core#2368]
(t) = testing branch.Our meta issue of #2362 has an entry for mostly failing dashboard. This is mostly likely down to the known failure of our data-collector system to start. This also affects numerous Web-UI 'live' header info. That to @FroggyFlox for doing some initial investigation into this issue that has arrisen post our recent Django update in the testing branch and soon to be testing channel releases.

From: less /opt/rockstor-dev/var/log/supervisord_data-collector_stderr.log

```
Traceback (most recent call last):
  File "/opt/rockstor-dev/bin/data-collector", line 31, in <module>
    import smart_manager.data_collector
  File "/opt/rockstor-dev/src/rockstor/smart_manager/data_collector.py", line 44, in <module>
    from system.pinmanager import (
  File "/opt/rockstor-dev/eggs/gevent-1.1.2-py2.7-linux-x86_64.egg/gevent/builtins.py", line 93, in __import__
    result = _import(*args, **kwargs)
  File "/opt/rockstor-dev/src/rockstor/system/pinmanager.py", line 23, in <module>
    from storageadmin.models import Pincard, EmailClient, User
  File "/opt/rockstor-dev/eggs/gevent-1.1.2-py2.7-linux-x86_64.egg/gevent/builtins.py", line 93, in __import__
    result = _import(*args, **kwargs)
  File "/opt/rockstor-dev/src/rockstor/storageadmin/models/__init__.py", line 19, in <module>
    from pool import Pool  # noqa E501
  File "/opt/rockstor-dev/eggs/gevent-1.1.2-py2.7-linux-x86_64.egg/gevent/builtins.py", line 93, in __import__
    result = _import(*args, **kwargs)
  File "/opt/rockstor-dev/src/rockstor/storageadmin/models/pool.py", line 34, in <module>
    class Pool(models.Model):
  File "/opt/rockstor-dev/eggs/Django-1.11.29-py2.7.egg/django/db/models/base.py", line 110, in __new__
    app_config = apps.get_containing_app_config(module)
  File "/opt/rockstor-dev/eggs/Django-1.11.29-py2.7.egg/django/apps/registry.py", line 247, in get_containing_app_config
    self.check_apps_ready()
  File "/opt/rockstor-dev/eggs/Django-1.11.29-py2.7.egg/django/apps/registry.py", line 125, in check_apps_ready
    raise AppRegistryNotReady("Apps aren't loaded yet.")
django.core.exceptions.AppRegistryNotReady: Apps aren't loaded yet.
```

**Comment by phillxnet** [comment]
And from the higher level supervisod log we have the perpetual spawned / exited data-collector with eventual timeout:
```
2022-04-11 17:10:35,577 CRIT Supervisor running as root (no user in config file)
2022-04-11 17:10:35,589 INFO RPC interface 'supervisor' initialized
2022-04-11 17:10:35,589 CRIT Server 'unix_http_server' running without any HTTP authentication checking
2022-04-11 17:10:35,590 INFO supervisord started with pid 14160
2022-04-11 17:10:36,593 INFO spawned: 'nginx' with pid 14165
2022-04-11 17:10:36,594 INFO spawned: 'gunicorn' with pid 14166
2022-04-11 17:10:36,596 INFO spawned: 'data-collector' with pid 14167
2022-04-11 17:10:36,598 INFO spawned: 'ztask-daemon' with pid 14168
2022-04-11 17:10:37,276 INFO exited: data-collector (exit status 1; not expected)
2022-04-11 17:10:38,428 INFO spawned: 'data-collector' with pid 14202
2022-04-11 17:10:38,756 INFO exited: data-collector (exit status 1; not expected)
2022-04-11 17:10:40,759 INFO spawned: 'data-collector' with pid 14208
2022-04-11 17:10:41,088 INFO exited: data-collector (exit status 1; not expected)
2022-04-11 17:10:42,089 INFO success: nginx entered RUNNING state, process has stayed up for > than 5 seconds (startsecs)
2022-04-11 17:10:42,089 INFO success: gunicorn entered RUNNING state, process has stayed up for > than 5 seconds (startsecs)
2022-04-11 17:10:44,093 INFO spawned: 'data-collector' with pid 14214
2022-04-11 17:10:44,421 INFO exited: data-collector (exit status 1; not expected)
2022-04-11 17:10:45,423 INFO gave up: data-collector entered FATAL state, too many start retries too quickly
2022-04-11 17:10:47,426 INFO success: ztask-daemon entered RUNNING state, process has stayed up for > than 10 seconds (startsecs)
2022-04-11 17:39:11,183 WARN received SIGTERM indicating exit request
2022-04-11 17:39:11,184 INFO waiting for nginx, gunicorn, ztask-daemon to die
2022-04-11 17:39:11,228 INFO exited: nginx (exit status 0; expected)
2022-04-11 17:39:11,231 INFO stopped: ztask-daemon (terminated by SIGTERM)
2022-04-11 17:39:11,262 INFO exited: gunicorn (exit status 0; expected)
```

**Comment by phillxnet** [comment]
Adding in the now required django.setup(), initially in data_collector.py, we have the following:

```
2022-04-11 18:44:54,422 CRIT Supervisor running as root (no user in config file)
2022-04-11 18:44:54,429 INFO RPC interface 'supervisor' initialized
2022-04-11 18:44:54,429 CRIT Server 'unix_http_server' running without any HTTP authentication checking
2022-04-11 18:44:54,430 INFO supervisord started with pid 23283
2022-04-11 18:44:55,433 INFO spawned: 'nginx' with pid 23287
2022-04-11 18:44:55,434 INFO spawned: 'gunicorn' with pid 23288
2022-04-11 18:44:55,437 INFO spawned: 'data-collector' with pid 23289
2022-04-11 18:44:55,440 INFO spawned: 'ztask-daemon' with pid 23290
2022-04-11 18:44:58,385 INFO success: data-collector entered RUNNING state, process has stayed up for > than 2 seconds (startsecs)
2022-04-11 18:45:01,389 INFO success: nginx entered RUNNING state, process has stayed up for > than 5 seconds (startsecs)
2022-04-11 18:45:01,390 INFO success: gunicorn entered RUNNING state, process has stayed up for > than 5 seconds (startsecs)
2022-04-11 18:45:06,397 INFO success: ztask-daemon entered RUNNING state, process has stayed up for > than 10 seconds (startsecs)
```

**Comment by phillxnet** [comment]
Closed by #2369
